in_toto/resolver/_resolver.py

```diff
@@ -205,3 +205,71 @@ def hash_artifacts(self, uris):
             os.chdir(original_cwd)
 
         return hashes
+
+
+class OSTreeResolver(Resolver):
+    """Resolver for OSTree repositories. """
+
+    SCHEME = "ostree"
+
+    # defining this separately as it should be pinned to OSTree's default
+    # rather than in-toto's
+    _HASH_ALGORITHM = "sha256"
+
+    def __init__(self, base_path=None):
+        self._base_path = base_path
+
+    def _strip_scheme_prefix(self, path):
+        """Helper to strip OSTree resolver scheme prefix from path."""
+
+        return path[len(self.SCHEME + ":") :]
+
+    def _add_scheme_prefix(self, path):
+        """Helper to add scheme back after recording the hash."""
+
+        return f"{self.SCHEME}:{path}"
+
+    def _hash(self, path):
+        """Helper to hash OSTree commits."""
+
+        ref_path = os.path.join("refs", "heads", path)
+        if not os.path.exists(ref_path):
+            raise OSTreeRefNotFound(f"OSTree ref '{path}' not found")
+
+        with open(ref_path, "r") as ref:  # pylint: disable=unspecified-encoding
+            ref_contents = ref.read()
+        ref_contents = ref_contents.strip("\n")
+
+        object_path = os.path.join("objects",
+                ref_contents[:2],
+                f"{ref_contents[2:]}.commit"
+        )
+
+        digest = digest_filename(object_path, algorithm=self._HASH_ALGORITHM)
+
+        return {self._HASH_ALGORITHM: digest.hexdigest()}
+
+    def hash_artifacts(self, uris):
+        hashes = {}
+
+        if self._base_path:
+            original_cwd = os.getcwd()
+            # FIXME: Re-raise for backwards-compat; should remove try/except
```

Comment:
No need to worry about backwards-compat in OSTreeResolver. :) I even changed this for FileResolver now in #592.